internal/cluster/validator.go

```diff
@@ -532,20 +545,8 @@ func (v *clusterValidator) printNoCidrsOverlapping(c *clusterPreprocessContext,
 	case ValidationSuccess:
 		return "No CIDRS are overlapping."
 	case ValidationFailure:
-		// TODO MGMT-7587: Support any number of subnets
-		// Assumes that the number of cluster networks equal to the number of service networks
-		for index := range c.cluster.ClusterNetworks {
-			machineNetworkCidr := ""
-			if len(c.cluster.MachineNetworks) > index {
-				machineNetworkCidr = string(c.cluster.MachineNetworks[index].Cidr)
-			}
-
-			if err := network.VerifyClusterCIDRsNotOverlap(machineNetworkCidr,
-				string(c.cluster.ClusterNetworks[index].Cidr),
-				string(c.cluster.ServiceNetworks[index].Cidr),
-				swag.BoolValue(c.cluster.UserManagedNetworking)); err != nil {
-				return fmt.Sprintf("CIDRS Overlapping: %s.", err.Error())
-			}
+		if err := checkCidrsOverlapping(c.cluster); err != nil {
```

Comment:
Why check again on ValidationFailure ?

Reply:
It's the code the already were there. I just put it into common function because fixes in the original flow did not propagate to this place. I guess that the original intent was to have different error messages in case of a failure

Reply:
In case you return ValidationError instead of ValidationFailure in other cases, then you will not have to rerun